// catalog/index.js
import React, { Component } from 'react'
import Image from 'semantic-ui-react/dist/commonjs/elements/Image'
import Button from 'semantic-ui-react/dist/commonjs/elements/Button'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getItems, viewItem, addToBasket } from 'actions/itemActions'

class Catalog extends Component {
    constructor(props) {
        super(props)
        this.state = {
            items: [],
            showModal: false
        }
    }
    componentDidMount() {
        this.props.get()
    }
    componentWillReceiveProps({ itemList, view, addToBasket }) {
        if (itemList !== this.state.items) {
            const items = itemList.map(item => {
                return (
                    <div className="item-box" key={item.id}>
                        <div className="item-grid">
                            <div className="item-desc-header">{item.name}</div>
                            <div className="item-image">
                                <Image src={item.image} />
                            </div>
                            <div className="item-desc">
                                <p className="item-desc-text" />
                            </div>
                            <div className="item-buttons">
                                <Button.Group vertical floated="right">
                                    <Button
                                        content="Подробнее"
                                        icon="search"
                                        labelPosition="left"
                                        onClick={() => view(item.id)}
                                    />
                                    <Button
                                        content="Добавить"
                                        icon="shop"
                                        labelPosition="left"
                                        onClick={() => addToBasket(item, 1)}
                                    />
                                </Button.Group>
                            </div>
                        </div>
                    </div>
                )
            })
            this.setState({
                items: items
            })
        }
    }
    render() {
        return <div className="item-container">{this.state.items}</div>
    }
}

const mapStateToProps = state => {
    return {
        itemList: state.items.list,
        showModal: state.items.showModal
    }
}

const mapDispatchToProps = dispatch => {
    return bindActionCreators(
        {
            get: getItems,
            view: viewItem,
            addToBasket: addToBasket
        },
        dispatch
    )
}

export default connect(mapStateToProps, mapDispatchToProps)(Catalog)
