// actions/itemActions.js
import {
    GET_ITEMS,
    VIEW_ITEM,
    VIEW_ITEM_CLOSE,
    ADD_TO_BASKET,
    REMOVE_FROM_BASKET,
    CHECKOUT
} from './types'
import axios from 'axios'
import { backendHost } from '../../api.config'

export const getItems = () => dispatch => {
    axios({
        method: 'get',
        baseURL: backendHost,
        url: '/api/items'
    }).then(res => {
        const items = res.data
        dispatch({
            type: GET_ITEMS,
            payload: items
        })
    })
}

export const viewItem = id => dispatch => {
    axios({
        method: 'get',
        baseURL: backendHost,
        url: '/api/items/' + id
    }).then(res => {
        const item = res.data
        dispatch({
            type: VIEW_ITEM,
            payload: item
        })
    })
}

export const viewItemClose = () => dispatch => {
    dispatch({
        type: VIEW_ITEM_CLOSE
    })
}

export const addToBasket = (item, count) => dispatch => {
    dispatch({
        type: ADD_TO_BASKET,
        payload: {
            item: item,
            count: count
        }
    })
}

export const removeFromBasket = id => dispatch => {
    dispatch({
        type: REMOVE_FROM_BASKET,
        payload: id
    })
}

export const checkout = basket => dispatch => {
    dispatch({
        type: CHECKOUT,
        payload: basket
    })
}
