// basket/checkout.js
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { checkout } from 'actions/itemActions'

class CheckoutButtons extends Component {
    constructor(props) {
        super(props)
        this.state = {
            page: 1,
            basket: this.props.basket
        }
    }
    static getDerivedStateFromProps(props, state) {
        if (props.page !== state.page)
            return {
                page: props.page
            }
        return state
    }
    switchPage = next => this.props.switchPage(next)
    render() {
        const { switchPage } = this
        const next = (
            <button onClick={() => this.switchPage(true)}>Дальше</button>
        )
        const back = (
            <button onClick={() => this.switchPage(false)}>Назад</button>
        )
        const checkout = (
            <button onClick={() => this.props.checkout()}>Checkout</button>
        )
        return (
            <div className="basket-checkout">
                {this.state.page > 1 && back}
                {this.state.page < 3 && next}
                {this.state.page == 3 && checkout}
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        basket: state.items.basket
    }
}
const mapDispatchToProps = dispatch => {
    return bindActionCreators(
        {
            checkout: checkout
        },
        dispatch
    )
}

export default connect(mapStateToProps, mapDispatchToProps)(CheckoutButtons)
