// components/dropdown/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'

class Dropdown extends Component {
    constructor(props) {
        super(props)
        this.state = {
            list: this.props.src,
            selected: 0,
            listOpen: false,
        }
    }
    handleSelect = (id) => {
        this.setState({
            selected: id,
            listOpen: !this.state.listOpen
        })
    }
    toggleList = () => {
        this.setState({
            listOpen: !this.state.listOpen
        })
    }
    render() {
        const {list, selected, listOpen} = this.state
        const {title, children} = this.props
        return (
            <div className="dd-wrapper">
                <div className="dd-header" onClick={() => this.toggleList()}>
                    <div className="dd-header-title">
                        {
                            selected?
                                list[selected-1].text : title?
                                title : children?
                                    children: '...'
                        }
                    </div>
                </div>
                {listOpen && <ul className="dd-list">
                    {list.map(item => {
                        let {text, id} = item
                        return (
                            <li className='dd-list-item' key={id} onClick={() => this.handleSelect(id)}>
                                {text}
                            </li>
                        )
                    })}
                </ul>}

            </div>
        )
    }
}

const mapStateToProps = state => {
    return {}
}

export default connect(
    mapStateToProps,
)(Dropdown)
